### PR TITLE
Handle empty output in mpl circuit drawer

### LINKED ISSUE #1845: Matplotlib circuit drawer raises IndexError if there is no gate in QuantumCircuit
### Information

- **Qiskit Terra version**: 0.7.3
- **Python version**: 3.6.8
- **Operating system**: macOS HighSierra

### What is the current behavior?

If I try to draw a quantum circuit without any gate, the mpl drawer raises IndexError. The text drawer does not have this issue.

### Steps to reproduce the problem

```
# sample.py
from qiskit import QuantumCircuit, QuantumRegister
qr = QuantumRegister(1)
qc = QuantumCircuit(qr)
print(qc)
qc.draw(filename='output.pdf', output='mpl')
```

```
$ python sample.py
         
q0_0: |0>
         
Traceback (most recent call last):
  File "c.py", line 5, in <module>
    qc.draw(filename='output.pdf', output='mpl')
  File "/Users/ima/envs/vqe2/lib/python3.6/site-packages/qiskit/circuit/quantumcircuit.py", line 353, in draw
    reverse_bits=reverse_bits)
  File "/Users/ima/envs/vqe2/lib/python3.6/site-packages/qiskit/tools/visualization/_circuit_visualization.py", line 237, in circuit_drawer
    reverse_bits=reverse_bits)
  File "/Users/ima/envs/vqe2/lib/python3.6/site-packages/qiskit/tools/visualization/_circuit_visualization.py", line 577, in _matplotlib_circuit_drawer
    return qcd.draw(filename)
  File "/Users/ima/envs/vqe2/lib/python3.6/site-packages/qiskit/tools/visualization/_matplotlib.py", line 343, in draw
    self._draw_ops(verbose)
  File "/Users/ima/envs/vqe2/lib/python3.6/site-packages/qiskit/tools/visualization/_matplotlib.py", line 466, in _draw_ops
    next_ops.pop(0)
IndexError: pop from empty list
```

### What is the expected behavior?

No IndexError.

### Suggested solutions

Check whether `next_ops` is empty or not.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

If an empty quantum circuit is passed into the draw function with the
mpl output specified it will stack trace. This is because internally the
mpl drawer assumes there will be always registers and operations that it
can iterate over. This commit fixes that so passing in an empty quantum
circuit is no longer fatal. It will return a blank white image, which is
consistent with the behavior of the latex and text output types.

### Details and comments

Fixes #1845

### CODE REVIEW

File: qiskit/tools/visualization/_matplotlib.py [hunk 1]
```diff
@@ -682,7 +683,11 @@ def _draw_ops(self, verbose=False):
         #
         # adjust window size and draw horizontal lines
         #
-        max_anc = max([q_anchors[ii].get_index() for ii in self._qreg_dict])
+        anchors = [q_anchors[ii].get_index() for ii in self._qreg_dict]
+        if anchors:
+            max_anc = max(anchors)
+        else:
+            max_anc = 0
         n_fold = (max_anc - 1) // self._style.fold
```
Comment (nkanazawa1989) [inline qiskit/tools/visualization/_matplotlib.py]: @mtreinish Can you change this line as follows, to show registers when empty circuit is given?
```
n_fold = max(0, max_anc - 1) // self._style.fold
```

Comment (mtreinish) [reply qiskit/tools/visualization/_matplotlib.py]: Ok, sure. It is updated with https://github.com/Qiskit/qiskit-terra/pull/1866/commits/2a74b13eff3156bc1b4293a03c49d19d3093fead

**Comment by mtreinish** [comment]
Just for completeness the output of a completely empty circuit (ie QuantumCircuit().draw(output='mpl') ) is:

out

and the output with a circuit with just registers (ie QuantumCircuit(QuantumRegister(3)).draw(output='mpl') ) is:

out2

**Comment by t-imamichi** [comment]
Looks good. Thank you!

**Comment by dcmckayibm** [comment]
This issue might have good overlap with this PR https://github.com/Qiskit/qiskit-terra/issues/1870

**Pull request merged**